src/components/UiTooltip/TooltipWrapper.tsx

```diff
@@ -0,0 +1,49 @@
+import {
+  ClickAwayListener,
+  Tooltip,
+  Typography,
+  useMediaQuery,
+} from '@mui/material';
+import React from 'react';
+
+import { UiTooltipProps } from './types';
+
+export default function WrapperUiTooltip({
+  children,
+  title,
+  placement,
+  arrow,
+  sx,
+}: UiTooltipProps): React.ReactElement {
+  const [open, setOpen] = React.useState(false);
+  const isWideScreenMaxWidth: boolean = useMediaQuery('(max-width: 640px)');
+  const isWideScreenMinWidth: boolean = useMediaQuery('(min-width: 640px)');
+
+  React.useEffect(() => {
+    if (isWideScreenMaxWidth || isWideScreenMinWidth) {
+      setOpen(false);
+    }
+  }, [isWideScreenMaxWidth, isWideScreenMinWidth]);
+
+  function handleTooltipClose(): void {
+    setOpen(false);
+  }
+  function handleTooltipToogle(): void {
+    setOpen(!open);
+  }
+  return (
+    <ClickAwayListener onClickAway={() => handleTooltipClose()}>
```

Comment:
why do you provide callback function? you can provide just the name of this function and this will be enough